cmd/nerdctl/container_restart_linux_test.go

```diff
@@ -51,11 +51,11 @@ func TestRestartPIDContainer(t *testing.T) {
 	base := testutil.NewBase(t)
 
 	baseContainerName := testutil.Identifier(t)
-	base.Cmd("run", "-d", "--name", baseContainerName, testutil.AlpineImage, "sleep", "infinity").Run()
+	base.Cmd("run", "-d", "--name", baseContainerName, testutil.AlpineImage, "sleep", "infinity").AssertOK()
 	defer base.Cmd("rm", "-f", baseContainerName).Run()
 
-	sharedContainerName := testutil.Identifier(t)
-	base.Cmd("run", "-d", "--name", sharedContainerName, fmt.Sprintf("--pid=container:%s", baseContainerName), testutil.AlpineImage, "sleep", "infinity").Run()
+	sharedContainerName := fmt.Sprintf("%s-shared", baseContainerName)
+	base.Cmd("run", "-d", "--name", sharedContainerName, fmt.Sprintf("--pid=container:%s", baseContainerName), testutil.AlpineImage, "sleep", "infinity").AssertOK()
```

Comment:
nit: since we are changing Run --> AssertOk; can we do the same for defer statements in this test as well.

Reply:
defer should not cause test to fail

Reply:
The reason for doing this in defer is to prevent Leaks and avoid flakiness; when a wrong command is written by mistake.